cmd/kubelet/app/options/options.go

```diff
@@ -163,7 +157,7 @@ func (s *KubeletServer) AddFlags(fs *pflag.FlagSet) {
 	fs.Int32Var(&s.HealthzPort, "healthz-port", s.HealthzPort, "The port of the localhost healthz endpoint")
 	fs.Var(componentconfig.IPVar{Val: &s.HealthzBindAddress}, "healthz-bind-address", "The IP address for the healthz server to serve on, defaulting to 127.0.0.1 (set to 0.0.0.0 for all interfaces)")
 	fs.Int32Var(&s.OOMScoreAdj, "oom-score-adj", s.OOMScoreAdj, "The oom-score-adj value for kubelet process. Values must be within the range [-1000, 1000]")
-	fs.BoolVar(&s.RegisterNode, "register-node", s.RegisterNode, "Register the node with the apiserver (defaults to true if --api-servers is set)")
+	fs.BoolVar(&s.RegisterNode, "register-node", s.RegisterNode, "Register the node with the apiserver. [default=true]")
```

Comment:
`pkg/apis/componentconfig/v1alpha1/defaults.go` sets the default to true. 

Based on:
`git grep RegisterNode`:
```
cmd/kubelet/app/options/options.go:     fs.BoolVar(&s.RegisterNode, "register-node", s.RegisterNode, "Register the node with the apiserver. [default=true]")
pkg/apis/componentconfig/types.go:      RegisterNode bool
pkg/apis/componentconfig/v1alpha1/defaults.go:  if obj.RegisterNode == nil {
pkg/apis/componentconfig/v1alpha1/defaults.go:          obj.RegisterNode = boolVar(true)
pkg/apis/componentconfig/v1alpha1/types.go:     RegisterNode *bool `json:"registerNode"`
Binary file pkg/apis/componentconfig/v1alpha1/zz_generated.conversion.go matches
Binary file pkg/apis/componentconfig/v1alpha1/zz_generated.deepcopy.go matches
pkg/kubelet/kubelet.go:         registerNode:                   kubeCfg.RegisterNode,
pkg/kubemark/hollow_kubelet.go: c.RegisterNode = true
test/e2e/framework/test_context.go:func RegisterNodeFlags() {
test/e2e_node/e2e_node_suite_test.go:   framework.RegisterNodeFlags()
```
the only `RegisterNode` usage after defaulting is assignment to a property of the internal `Kubelet` object.

And based on `git grep registerNode`:
```
pkg/apis/componentconfig/types.go:      // registerNode enables automatic registration with the apiserver.
pkg/apis/componentconfig/types.go:      // the kubelet registers itself. This only takes effect when registerNode
pkg/apis/componentconfig/v1alpha1/types.go:     // registerNode enables automatic registration with the apiserver.
pkg/apis/componentconfig/v1alpha1/types.go:     RegisterNode *bool `json:"registerNode"`
pkg/apis/componentconfig/v1alpha1/types.go:     // the kubelet registers itself. This only takes effect when registerNode
Binary file pkg/generated/openapi/zz_generated.openapi.go matches
pkg/kubelet/kubelet.go:         registerNode:                   kubeCfg.RegisterNode,
pkg/kubelet/kubelet.go: registerNode bool
pkg/kubelet/kubelet_node_status.go:     if kl.registerNode {
```
This appears to be the only assignment to that object.
The only usage of that property is in `pkg/kubelet/kubelet_node_status.go`, in `syncNodeStatus`. 

Is there something else that could be defaulting it to false?